src/tests/eolian/eolian_generation.c

```diff
@@ -65,18 +65,89 @@ _files_compare (const char *ref_filename, const char *tmp_filename)
    return result;
 }
 
+/**
+ * Returns 0 on success, -1 on failure.
+ */
+static int
+_create_response_file(const char *rsp_path, const char *eo_filename, const char *type, const char *output_filename)
+{
+    printf("Opening %s...", rsp_path);
+
+    remove(rsp_path);
+    FILE *f = fopen(rsp_path, "w");
+
+    if (!f)
+    {
+        printf("Failed!\n");
+        return -1;
+    }
+    printf("Done!\n");
+
+    printf("Writing to file...\n"
+        "    :: -S\n"
+        "    :: -I \""TESTS_SRC_DIR"\"\n"
+        "    :: -o %s:%s\n"
+        "    :: %s\n",
+        type, output_filename,
+        eo_filename
+    );
+
+    fprintf(f,
+        "-S\n"
+        "-I\n"
+        "\""TESTS_SRC_DIR"\"\n"
+        "-o\n"
+        "%s:%s\n"
+        "%s\n",
+        type, output_filename,
+        eo_filename
+    );
+
+    if (ferror(f))
+        printf("Foi no ferror");
+
+    if (fclose(f))
+        return -1;
+
+    printf("Done successfully!\n");
+
+    return 0;
+}
+
 static int
 _eolian_gen_execute(const char *eo_filename, const char *type, const char *output_filename)
 {
    char command[PATH_MAX];
+
    if (snprintf(command, PATH_MAX,
                 EOLIAN_GEN" -S -I \""TESTS_SRC_DIR"/data\" -o %s:%s %s",
                 type, output_filename, eo_filename) > PATH_MAX)
      {
-        printf("eolian gen command too long for buffer\n");
-        abort();
+        printf("eolian gen command too long for buffer. Fallbacking to a response file\n");
+
+        char rsp_path[PATH_MAX];
+        snprintf(rsp_path, PATH_MAX, "%s.rsp", output_filename);
+
+        snprintf(command, PATH_MAX,
+                 EOLIAN_GEN" -r \"%s\"",
+                 rsp_path);
+
+        int failed = _create_response_file(rsp_path, output_filename, type, eo_filename);
+
+        if (failed)
+        {
+            fprintf(stderr, "Failed to create response file %s\n", rsp_path);
+            perror("Reason");
+            fflush(stderr);
+            return -1;
```

Comment:
Are `abort` and `return -1` equivalent semanticaly?

Reply:
`abort` is to send a `SIGABORT` to the program. `-1` is just a "things went wrong!" answer to the calling function.